### DISCUSSION TITLE
Micronaut Data JDBC ManyToMany Problem

### DISCUSSION THREAD

**Discussion by sebplorenz** [micronaut-projects/micronaut-data#1167] [Q&A] [answered]
Not sure if this is a bug or if I'm using it wrong. I have an issue with a ManyToMany Relation between two entities. The relation is created correctly, but never deleted.
I created a sample application with two entities A and B. They have a many-to-many relation to each other.

```
@Data
@MappedEntity
public class A {

    @Id
    String id;

    String value;

    @JoinTable(name = "ab")
    @Relation(value = Relation.Kind.MANY_TO_MANY, cascade = Relation.Cascade.ALL)
    Set<B> bs;

}

@Data
@MappedEntity
public class B {

    @Id
    String id;

    String value;

    @Relation(value = Relation.Kind.MANY_TO_MANY, mappedBy = "bs")
    Set<A> as;

}
```
I'm able to connect an instance of B to A with the following code:
```
    @Transactional(Transactional.TxType.REQUIRES_NEW)
    void setup() {
        log.info("Create a");
        var a = new A().setId("a").setValue("test");
        a = aRepository.save(a);

        log.info("Create b");
        var b = new B().setId("b").setValue("test");
        b = bRepository.save(b);

        log.info("Update a with b");
        Set<B> bs = new HashSet<>();
        bs.add(b);
        a.setBs(bs);
        aRepository.update(a);
    }
```
The following code should (imho) remove the relation between A and B:
```
    @Transactional(Transactional.TxType.REQUIRES_NEW)
    void deleteBFromA() {
        log.info("Remove b from a");
        var a = aRepository.findById("a").orElseThrow();
        a.getBs().clear();
        aRepository.update(a);

        var b = bRepository.findById("b").orElseThrow();
        b.getAs().clear();
        bRepository.update(b);
    }
```
But b is not removed from a. In the SQL logs I can see that table "a" and table "b" are updated. But the join table "ab" is not updated. What am I doing wrong?

Btw. I'm using Micronaut 3.0.1. My example can be found here: https://github.com/sebplorenz/demo-jdbc-micronaut3

**Comment by dstepanov** [comment, marked as answer]
We don't support it as there is no way to tell which items have been added or deleted.

**Reply by graemerocher** [reply]
As @dstepanov said Micronaut Data JDBC is not an ORM / object synchronization framework but instead a simple data mapper if you are interested in that you should use Micronaut Data JPA otherwise write your own logic to synchronize the state

**Reply by sebplorenz** [reply]
Thanks for the reply @dstepanov and @graemerocher. I think I'll go with the Micronaut Data JPA solution.
